dialoptions.go

```diff
@@ -712,6 +714,19 @@ func WithIdleTimeout(d time.Duration) DialOption {
 	})
 }
 
+// WithMaxRetryAttempts returns a DialOption that configures the maximum number
+// of retry attempts for the channel. Service owners may specify a higher value
+// for these parameters, but higher values will be treated as equal to the
+// maximum value by the client implementation. This mitigates security concerns
+// related to the service config being transferred to the client via DNS.
+//
+// A default value of 5 will be used if this dial option is not set.
+func WithMaxRetryAttempts(n int) DialOption {
+	return newFuncDialOption(func(o *dialOptions) {
+		o.maxRetryAttempts = n
```

Comment:
Are there no bounds that should be placed on this?

The `maxAttempts` field in the retry config must be >1, so that seems like a reasonable check for this as well.  (There's a separate setting to disable retry entirely.)

Since we can't return an error, maybe:

```
// A default value of 5 will be used if this dial option is not set or if n < 2.
```

?

Reply:
The spec does not place any upper bound on it.

I think your comment and logic make sense. I'll update it.